batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,422 @@
+		const char *extra_input_files, const char *extra_output_files, \
+		struct jx *envlist, const struct rmsummary *resources )
+{
+
+	// Get the path to mesos python site-packages
+	if (is_mesos_py_path_known != 1) {
+		mesos_py_path = batch_queue_get_option(q, "mesos-path");
+		if (mesos_py_path == NULL) {
+			debug(D_ERROR, "Please set the path to mesos python library\n");
+			exit(EXIT_FAILURE);
+		} else {
+			debug(D_INFO, "Get mesos_path %s from command line\n", mesos_py_path);
+			is_mesos_py_path_known = 1;
+		}
+	}
+
+	// Get the mesos master address
+	if (is_mesos_master_known != 1) {
+		mesos_master = batch_queue_get_option(q, "mesos-master");
+		if (mesos_master == NULL) {
+			debug(D_ERROR, "Please specify the ip address of mesos master\n");
+			exit(EXIT_FAILURE);
+		} else {
+			debug(D_INFO, "Get mesos_path %s from command line\n", mesos_py_path);
+			is_mesos_master_known = 1;
+		}
+	}
+
+	if (is_mesos_py_path_known == 1 && is_mesos_master_known == 1 &&is_scheduler_running == 0) {
+		start_mesos_scheduler(q);
+		is_scheduler_running = 1;
+	}
+
+	int task_id = ++counter;
+
+	debug(D_BATCH, "task %d is ready", task_id);
+	struct batch_job_info *info = malloc(sizeof(*info));
+	memset(info, 0, sizeof(*info));
+	info->submitted = time(0);
+	info->started = time(0);
+	itable_insert(q->job_table, task_id, info);
+
+	FILE *task_info_fp;
+
+	if(access(FILE_TASK_INFO, F_OK) != -1) {
+		task_info_fp = fopen(FILE_TASK_INFO, "a+");
+	} else {
+		task_info_fp = fopen(FILE_TASK_INFO, "w+");
+	}
+	
+	struct mesos_task *mt = mesos_task_create(task_id, cmd, extra_input_files, extra_output_files);
+
+	fprintf(task_info_fp, "%d,%s,", mt->task_id, mt->task_cmd);
+
+	// Get the absolut path o Meanwhile, the makeflow mesos scheduler keep polling 
+	// mesos_task_info, try to find if there is new task is ready to launch on mesos,. 
+	// 
+	// After the task is complete on, the makeflow mesos scheduler writes the final state
+	// of the task to mesos_task_state. And the makeflow is keep monitoring this file and 
+	// collect all finished task. f each input file
+
+	if (extra_input_files != NULL && strlen(extra_input_files) != 0) {
+
+		int j = 0;
+		int num_input_files = mt->task_input_files->used_length;
+		for (j = 0; j < (num_input_files-1); j++) {
+			fprintf(task_info_fp, "%s ", (mt->task_input_files->items)[j]);
+		}
+		fprintf(task_info_fp, "%s,", (mt->task_input_files->items)[num_input_files-1]);
+
+	} else {
+		fputs(",", task_info_fp);
+	}
+    
+	if (extra_output_files != NULL && strlen(extra_output_files) != 0) {
+		int j = 0;
+		int num_output_files = mt->task_output_files->used_length;
+		for (j = 0; j < (num_output_files-1); j++) {
+			fprintf(task_info_fp, "%s ", (mt->task_output_files->items)[j]);
+		}
+		fprintf(task_info_fp, "%s,", (mt->task_output_files->items)[num_output_files-1]);
+	} else {
+		fputs(",", task_info_fp);
+	}
+
+	int64_t cores = 4;
```

Comment:
There is no specific reason for these defaults, they are just for testing resource monitor. I have set it back to `cores = 1, memory = 1024, disk = 1024`